app/src/main/java/com/example/sharingang/utils/PermissionRequest.kt

```diff
@@ -0,0 +1,76 @@
+package com.example.sharingang.utils
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.pm.PackageManager
+import android.location.Location
+import android.widget.Toast
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationRequest
+import com.google.android.gms.tasks.CancellationTokenSource
+
+fun requestPermissionLauncher(
+    fragment: Fragment,
+    callback: () -> Unit
+): ActivityResultLauncher<String> {
+    return fragment.registerForActivityResult(ActivityResultContracts.RequestPermission()) { isGranted: Boolean ->
+        if (isGranted) {
+            callback()
+        }
```

Comment:
```suggestion
    return fragment.registerForActivityResult(ActivityResultContracts.RequestPermission()) {
        if (it) {
            callback()
        }
```

I think this works?

Reply:
I just thought "isGranted" would be more understandable as to what this method does

Reply:
fair enough